Translate concepts/security/overview.md in Korean (#19464)
kubernetes/website: content/ko/docs/concepts/security/overview.md

주요 공격 영역 중 하나이다. 이런 코드 수준은 쿠버네티스의 범위 
밖이지만 몇가지 권장사항이 있다.

### 일반적인 코드 보안 지침표

코드에서 고려할 영역 | 추천 |
--------------------------------------------- | ------------ |
TLS를 통한 접근 | 코드가 TCP를 통해 통신해야 한다면, 클라이언트와 먼저 TLS 핸드 셰이크를 수행하는 것이 이상적이다. 몇 가지 경우를 제외하고, 기본 동작은 전송 중인 모든 것을 암호화하는 것이다. 한걸음 더 나아가, VPC의 "방화벽 뒤"에서도 서비스 간 네트워크 트래픽을 암호화하는 것이 좋다. 이것은 인증서를 가지고 있는 두 서비스의 양방향 검증을 [mTLS](https://en.wikipedia.org/wiki/Mutual_authentication)를 통해 수행할 수 있다. 이것을 수행하기 위해 쿠버네티스에는 [Linkerd](https://linkerd.io/) 및 [Istio](https://istio.io/)와 같은 수많은 도구가 있다. |
통신 포트 범위 제한 | 이 권장사항은 당연할 수도 있지만, 가능하면 통신이나 메트릭 수집에 꼭 필요한 서비스의 포트만 노출시켜야 한다. |
타사 종속성 보안 | 애플리케이션은 자체 코드베이스의 외부에 종속적인 경향이 있기 때문에, 코드의 종속성을 정기적으로 스캔하여 현재 알려진 취약점이 없는지 확인하는 것이 좋다. 각 언어에는 이런 검사를 자동으로 수행하는 도구를 가지고 있다. |
정적 코드 분석 | 대부분 언어에는 잠재적으로 안전하지 않은 코딩 방법에 대해 코드 스니펫을 분석할 수 있는 방법을 제공한다. 가능한 언제든지 일반적인 보안 오류에 대해 코드베이스를 스캔할 수 있는 자동화된 도구를 사용하여 검사를 한다. 도구는 다음에서 찾을 수 있다: https://www.owasp.org/index.php/Source_Code_Analysis_Tools |
동적 탐지 공격 | 일반적으로 서비스에서 발생할 수 있는 잘 알려진 공격 중 일부를 서비스에 테스트할 수 있는 자동화된 몇 가지 도구가 있다. 이런 잘 알려진 공격에는 SQL 인젝션, CSRF 및 XSS가 포함된다. 가장 널리 사용되는 동적 분석 도구는 OWASP Zed Attack 프록시다. https://www.owasp.org/index.php/OWASP_Zed_Attack_Proxy_Project |

> ## 강력한 자동화

Reviewer: ```suggestion
## 강력(robust)한 자동화
```
robust 는 보통 내성이 높은 것을 의미하는 것 같습니다. "강력한 자동화"는 어색한 것 같아서, 기존 영문을 병기하여 보완하면 좋을 것 같습니다.

위에서 언급한 대부분의 제안사항은 실제로 일련의 보안 검사의 일부로 코드를 

> 전달하는 파이프라인에 의해 자동화 될 수 있다. 소프트웨어 전달에 대한 

Reviewer: ```suggestion
전달하는 파이프라인에 의해 자동화 될 수 있다. 소프트웨어 전달을 위한 
```
문장에 "대한" 이 반복적으로 사용되어, 자연스럽게 변경해 보았습니다. 참고하시기 바랍니다.

> "지속적인 해킹(Continuous Hacking)"에 대한 접근 방식에 대해 알아 보려면 이 [기사](https://thenewstack.io/beyond-ci-cd-how-continuous-hacking-of-docker-containers-and-pipeline-driven-security-keeps-ygrene-secure/)에서 자세히 설명하고 있다.

Reviewer: ```suggestion
"지속적인 해킹(Continuous Hacking)"에 대한 접근 방식에 대해 알아 보려면, 자세한 설명을 제공하는 [이 기사](https://thenewstack.io/beyond-ci-cd-how-continuous-hacking-of-docker-containers-and-pipeline-driven-security-keeps-ygrene-secure/)를 본다.
```
조금 자연스럽게 변경해 보았습니다.

{{% /capture %}}
{{% capture whatsnext %}}
* [파드에 대한 네트워크 정책](/ko/docs/concepts/services-networking/network-policies/) 알아보기
* [클러스터 보안](/ko/docs/tasks/administer-cluster/securing-a-cluster/)에 대해 알아보기
* [API 접근 통제](/ko/docs/reference/access-authn-authz/controlling-access/)에 대해 알아보기
* 컨트롤 플레인에 대한 [전송 데이터 암호화](/docs/tasks/tls/managing-tls-in-a-cluster/) 알아보기
* [Rest에서 데이터 암호화](/ko/docs/tasks/administer-cluster/encrypt-data/) 알아보기

> * [쿠버네티스 시크릿](/ko/docs/concepts/configuration/secret/)에 대해 알아보기

Reviewer: 현재 한글화가 완료된 문서에 한해서 링크를 한글 문서로 변경하고 있습니다.
확인해보니, 해당 라인에 대한 문서들은 한글문서가 없습니다. 기존 영문 문서의 링크로 변경이 필요합니다.
```suggestion
* [클러스터 보안](/docs/tasks/administer-cluster/securing-a-cluster/)에 대해 알아보기
* [API 접근 통제](/docs/reference/access-authn-authz/controlling-access/)에 대해 알아보기
* 컨트롤 플레인에 대한 [전송 데이터 암호화](/docs/tasks/tls/managing-tls-in-a-cluster/) 알아보기
* [Rest에서 데이터 암호화](/docs/tasks/administer-cluster/encrypt-data/) 알아보기
* [쿠버네티스 시크릿](/docs/concepts/configuration/secret/)에 대해 알아보기
```

---
title: 클라우드 네이티브 보안 개요
content_template: templates/concept
weight: 1
---

{{< toc >}}

{{% capture overview %}}
쿠버네티스 보안(일반적인 보안)은 관련된 많은 부분이 상호작용하는 
방대한 주제다. 오늘날에는 웹 애플리케이션의 실행을 돕는 
수많은 시스템에 오픈소스 소프트웨어가 통합되어 있으며, 
전체적인 보안에 대하여 생각할 수 있는 방법에 대한 통찰력을 도울 수 있는 

> 몇가지 중요한 개념이 있다. 이 가이드는 클라우드 네이티브 보안과 관련된 

Reviewer 2: ```suggestion
전체적인 보안에 대하여 생각할 수 있는 방법에 대한 통찰력을 도울 수 있는 
몇 가지 중요한 개념이 있다. 이 가이드는 클라우드 네이티브 보안과 관련된 
```

사물의 수나 양을 나타내어 체언을 꾸미는 수관형사는 뒤에 오는 명사 또는 의존 명사와 띄어 씁니다.

몇가지 일반적인 개념에 대한 멘탈 모델을 정의한다. 멘탈 모델은 완전히 임의적이며 

> 소프트웨어 스택을 보호할 위치를 생각하는데 도움이되는 경우에만 사용해야 

Reviewer 2: ```suggestion
몇 가지 일반적인 개념에 대한 멘탈 모델을 정의한다. 멘탈 모델은 완전히 임의적이며 
소프트웨어 스택을 보호할 위치를 생각하는데 도움이 되는 경우에만 사용해야 
```

도움이 되는 : 조사나 어미 다음에 오는 말은 띄어 써야 합니다.

한다.
{{% /capture %}}

{{% capture body %}}

## 클라우드 네이티브 보안의 4C

> 계층적인 보안에 대해서 어떻게 생각할 수 있는지 이해하는데 도움이 될 수 있는 다이어그램 부터 살펴보자.

Reviewer 2: ```suggestion
계층적인 보안에 대해서 어떻게 생각할 수 있는지 이해하는 데 도움이 될 수 있는 다이어그램부터 살펴보자.
```

- 이해하는 데 : '데'가 '곳'이나 '장소', '일'이나 '것', '경우'의 뜻을 나타낼 때는 의존명사로 띄어 써야 합니다.
- 다이어그램부터 : 조사이므로 앞 어절과 붙여야 합니다.

{{< note >}}

> 이 계층화된 접근 방식은 보안에 대한 [심층 방어](https://en.wikipedia.org/wiki/Defense_in_depth_(computing)) 

Reviewer 3: ```suggestion
이 계층화된 접근 방식은 보안에 대한 [심층 방어](https://ko.wikipedia.org/wiki/%EC%A2%85%EC%8B%AC%EB%B0%A9%EC%96%B4)
```
?

Author: That link(ko.) is about military strategy of category, not about computing.
There are no Korean translated page about "defense in depth" with computing.

Reviewer 4: If there's no matching page in Korean wikipedia, how about leaving the original link? Different content may cause misunderstanding.

Author: @gochist
Yes, left the original link.

접근 방식을 강화하며, 소프트웨어 시스템의 보안을 위한 모범 사례로 

> 널리 알려져 있다. 4C는 클라우드(Cloud), 클러스터(Clusters), 컨테이너(Containers) 및 코드(Code)이다.

Reviewer 3: [The “4C” thing might not be important to retain in this localization, as the Korean words don't all start with the same syllabic block]

Author: You're right. But I think Koreans can fully understand "4C".

{{< /note >}}

{{< figure src="/images/docs/4c.png" title="클라우드 네이티브 보안의 4C" >}}

위 그림에서 볼 수 있듯이,
4C는 각각의 사각형의 보안에 따라 다르다. 코드 

> 수준의 보안만 처리하여 클라우드, 컨테이너 및 코드의 열약한 보안 표준으로부터 

Reviewer 2: ```suggestion
수준의 보안만 처리하여 클라우드, 컨테이너 및 코드의 열악한 보안 표준으로부터 
```

오타로 보입니다.

> 보호하는 것은 거의 불가능 하다. 그러나 이런 영역들의 보안이 적절하게 

Reviewer 2: ```suggestion
보호하는 것은 거의 불가능하다. 그러나 이런 영역들의 보안이 적절하게 
```

'-하다'는 앞의 명사와 붙여야 합니다.

처리되고, 코드에 보안을 추가한다면 이미 강력한 기반이 더욱 

> 강화된다. 이러한 관심분야는 아래에서 더 자세히 설명한다.

Reviewer 2: ```suggestion
처리되고, 코드에 보안을 추가한다면 이미 강력한 기반이 더욱 
강화될 것이다. 이러한 관심 분야는 아래에서 더 자세히 설명한다.
```

좀 더 자연스럽게 의역해봤습니다.

## 클라우드

클라우드(또는 공동 위치 서버, 또는 기업의 데이터센터)는 쿠버네티스 클러스터의 다양한 
[신뢰 컴퓨팅 기반](https://en.wikipedia.org/wiki/Trusted_computing_base)
이다. 이러한 구성 요소 자체가 취약하거나(또는 취약한 방법으로 구성된) 
경우 이 기반 위에서 구축된 모든 구성 요소의 보안을 
실제로 보장할 방법이 없다. 각 클라우드 공급자는 그들의 환경에서 워크로드를 
안전하게 실행하는 방법에 대해 고객에게 다양한 보안 권장 사항을 
제공한다. 모든 클라우드 공급자와 워크로드는 다르기 때문에 
클라우드 보안에 대한 권장 사항을 제공하는 것은 이 가이드의 범위를 벗어난다. 쿠버네티스 
클러스터를 구성하는 인프라 보안에 대한 
클라우드 공급자의 문서 및 일반적인 지침에 대한 
링크가 있다.

> ### 클라우드 공급자 보안 테이블

Author: 표로 통일 하였습니다.

IaaS 공급자        | 링크 |
-------------------- | ------------ |
Alibaba Cloud | https://www.alibabacloud.com/trust-center |
Amazon Web Services | https://aws.amazon.com/security/ |
Google Cloud Platform | https://cloud.google.com/security/ |
IBM Cloud | https://www.ibm.com/cloud/security |
Microsoft Azure | https://docs.microsoft.com/en-us/azure/security/azure-security |
VMWare VSphere | https://www.vmware.com/security/hardening-guides.html |

자체 하드웨어나 다른 클라우드 공급자를 사용하는 경우 보안에 대한 
모범 사례는 해당 문서를 참조한다.

### 일반적인 인프라 지침 표

쿠버네티스 인프라에서 고려할 영역 | 추천 |
--------------------------------------------- | ------------ |
API 서버에 대한 네트워크 접근 (마스터) | 쿠버네티스 마스터에 대한 모든 접근은 공개적인 공개적인 인터넷에서 허용되지 않으며 클러스터를 관리하는데 필요한 IP 주소 범위로 제한된 네트워크 접근 제어 목록에 의해 제어 되어야 한다. |

> 노드에 대한 네트워크 접근 (워커 서버) | 마스터의 지정된 포트로 네트워크 접근 제어 목록을 통한 연결 _만_ 허용하고, 쿠버네티스의 NodePort 및 LoadBalancer 타입의 서비스 연결을 허용하도록 노드를 구성한다. 가능한 이런 노드가 공용 인터넷에 완전히 노출되어서는 안된다. 

Author: 원문에 ```_only_``` 라는 이태릭체가 되어 있어서 똑같이 한다고 ```_만_``` 이라고 했는데 안되는 건가요?

> 노드에 대한 네트워크 접근 (워커 서버) | 마스터의 지정된 포트로 네트워크 접근 제어 목록을 통한 연결 _만_ 허용하고, 쿠버네티스의 NodePort 및 LoadBalancer 타입의 서비스 연결을 허용하도록 노드를 구성한다. 가능한 이런 노드가 공용 인터넷에 완전히 노출되어서는 안된다. 

Reviewer 5: @c1t1d0s7 님 제가 잘못 보았던 것 같습니다.
`_만_` 표기 하는게 맞습니다 :) 참고하셔서 업데이트 부탁드립니다!

> 노드에 대한 네트워크 접근 (워커 서버) | 마스터의 지정된 포트로 네트워크 접근 제어 목록을 통한 연결 _만_ 허용하고, 쿠버네티스의 NodePort 및 LoadBalancer 타입의 서비스 연결을 허용하도록 노드를 구성한다. 가능한 이런 노드가 공용 인터넷에 완전히 노출되어서는 안된다. 

Author: @ysyukr 네, 확인 감사합니다.

쿠버네티스 인그레스를 위한 TLS | https://kubernetes.io/docs/concepts/services-networking/ingress/#tls

## 컨테이너

쿠버네티스에서 소프트웨어를 실행하려면, 소프트웨어는 컨테이너에 있어야한다. 이로 인해, 
쿠버네티스의 워크로드 보안의 기본 요소를 활용하기 위해 
고려해야할 특정 보안 고려사항이 있다. 컨테이너 보안 
또한 이 가이드의 범위를 벗어나지만, 주제에 대한 추가적인 
일반 권장사항 및 링크 표가 있다.

컨테이너에서 고려할 영역 | 추천 |
------------------------------ | ------------ |
컨테이너 취약점 스캔 및 OS에 의존적인 보안 | 이미지 빌드 단계의 일부 또는 정기적으로 [CoreOS의 Clair](https://github.com/coreos/clair/)와 같은 도구를 사용하여 알려진 취약점에 대해 컨테이너를 스캔한다.
이미지 서명 및 시행 | 두개의 다른 CNCF 프로젝트 (TUF 및 Notary)는 컨테이너 이미지에 서명하고 컨테이너 내용에 대한 신뢰 시스템을 유지하는데 유용한 도구이다. 도커를 사용하는 경우 도커 엔진에 [Docker Content Trust](https://docs.docker.com/engine/security/trust/content_trust/)가 내장되어 있다. 시행 부분에서의 [IBM의 Portieris](https://github.com/IBM/portieris) 프로젝트는 쿠버네티스 다이나믹 어드미션 컨트롤러로 실행되는 도구로, 클러스터에서 허가하기 전에 Notary를 통해 이미지가 적절하게 서명되었는지 확인한다.
권한있는 사용자의 비허용 | 컨테이너를 구성할 때 컨테이너의 목적을 수행하는데 필요한 최소 권한을 가진 사용자를 컨테이너 내에 만드는 방법에 대해서는 설명서를 참조한다.

## 코드

마지막으로 애플리케이션의 코드 수준으로 내려가면, 가장 많은 제어를 할 수 있는 
주요 공격 영역 중 하나이다. 이런 코드 수준은 쿠버네티스의 범위 
밖이지만 몇가지 권장사항이 있다.

### 일반적인 코드 보안 지침표

코드에서 고려할 영역 | 추천 |
--------------------------------------------- | ------------ |

> TLS를 통한 접근 | 코드가 TCP를 통해 통신해야 한다면, 클라이언트와 먼저 TLS 핸드 셰이크를 수행하는것이 이상적이다. 몇 가지 경우를 제외하고, 기본 동작은 전송중인 모든것을 암호화하는 것이다. 한걸음 더 나아가, VPC의 "방화벽 뒤"에서도 서비스 간 네트워크 트래픽을 암호화하는 것이 좋다. 이것은 인증서를 가지고 있는 두 서비스의 양방향 검증을 [mTLS](https://en.wikipedia.org/wiki/Mutual_authentication)를 통해 수행할 수 있다. 이것을 수행하기 위해 쿠버네티스에는 [Linkerd](https://linkerd.io/) 및 [Istio](https://istio.io/)와 같은 수많은 도구가 있다. |

Reviewer 5: ```suggestion
TLS를 통한 접근 | 코드가 TCP를 통해 통신해야 한다면, 클라이언트와 먼저 TLS 핸드 셰이크를 수행하는 것이 이상적이다. 몇 가지 경우를 제외하고, 기본 동작은 전송 중인 모든 것을 암호화하는 것이다. 한걸음 더 나아가, VPC의 "방화벽 뒤"에서도 서비스 간 네트워크 트래픽을 암호화하는 것이 좋다. 이것은 인증서를 가지고 있는 두 서비스의 양방향 검증을 [mTLS](https://en.wikipedia.org/wiki/Mutual_authentication)를 통해 수행할 수 있다. 이것을 수행하기 위해 쿠버네티스에는 [Linkerd](https://linkerd.io/) 및 [Istio](https://istio.io/)와 같은 수많은 도구가 있다. |
```

통신 포트 범위 제한 | 이 권장사항은 당연할 수도 있지만, 가능하면 통신이나 메트릭 수집에 꼭 필요한 서비스의 포트만 노출시켜야 한다. |
타사 종속성 보안 | 애플리케이션은 자체 코드베이스의 외부에 종속적인 경향이 있기 때문에, 코드의 종속성을 정기적으로 스캔하여 현재 알려진 취약점이 없는지 확인하는 것이 좋다. 각 언어에는 이런 검사를 자동으로 수행하는 도구를 가지고 있다. |
정적 코드 분석 | 대부분 언어에는 잠재적으로 안전하지 않은 코딩 방법에 대해 코드 스니펫을 분석할 수 있는 방법을 제공한다. 가능한 언제든지 일반적인 보안 오류에 대해 코드베이스를 스캔할 수 있는 자동화된 도구를 사용하여 검사를 한다. 도구는 다음에서 찾을 수 있다: https://www.owasp.org/index.php/Source_Code_Analysis_Tools |

> 동적 탐지 공격 | 일반적으로 서비스에서 발생할 수 있는 잘 알려진 공격 중 일부를 서비스에 테스트할 수 있는 자동화된 몇가지 도구가 있다. 이런 잘 알려진 공격에는 SQL 인젝션, CSRF 및 XSS가 포함된다. 가장 널리 사용되는 동적 분석 도구는 OWASP Zed Attack 프록시다. https://www.owasp.org/index.php/OWASP_Zed_Attack_Proxy_Project |

Reviewer 5: ```suggestion
동적 탐지 공격 | 일반적으로 서비스에서 발생할 수 있는 잘 알려진 공격 중 일부를 서비스에 테스트할 수 있는 자동화된 몇 가지 도구가 있다. 이런 잘 알려진 공격에는 SQL 인젝션, CSRF 및 XSS가 포함된다. 가장 널리 사용되는 동적 분석 도구는 OWASP Zed Attack 프록시다. https://www.owasp.org/index.php/OWASP_Zed_Attack_Proxy_Project |
```

## 강력한 자동화

위에서 언급한 대부분의 제안사항은 실제로 일련의 보안 검사의 일부로 코드를 
전달하는 파이프라인에 의해 자동화 될 수 있다. 소프트웨어 전달에 대한 

> "연속적인 해킹(Continuous Hacking)"에 대한 접근 방식에 대해 알아 보려면 이 [기사](https://thenewstack.io/beyond-ci-cd-how-continuous-hacking-of-docker-containers-and-pipeline-driven-security-keeps-ygrene-secure/)에서 자세히 설명하고 있다.

Reviewer 5: ```suggestion
"지속적인 해킹(Continuous Hacking)"에 대한 접근 방식에 대해 알아 보려면 이 [기사](https://thenewstack.io/beyond-ci-cd-how-continuous-hacking-of-docker-containers-and-pipeline-driven-security-keeps-ygrene-secure/)에서 자세히 설명하고 있다.
```

`Continuous` 에 대해 `지속적인` 으로 더 많이 통용되고 있어 변경해보았습니다.
